libs/deck-layers/src/layers/fourwings/FourwingsHeatmapTileLayer.ts

```diff
@@ -235,29 +233,31 @@ export class FourwingsHeatmapTileLayer extends CompositeLayer<
     if (isStartOutRange || isEndOutRange || interval !== this.state.interval) {
       this.setState({
         interval,
-        ...this.getCacheRange(minFrame, maxFrame),
+        ...this._getCacheRange(minFrame, maxFrame),
       })
     }
     return [this.state.cacheStart, this.state.cacheEnd, interval].join('-')
   }
 
-  _updateSublayerColorRanges = () => {
-    const { sublayers, colorRampWhiteEnd } = this.props
-    const newSublayerColorRanges = sublayers.map(({ config }) =>
-      HEATMAP_COLOR_RAMPS[
-        (colorRampWhiteEnd ? `${config.colorRamp}_toWhite` : config.colorRamp) as ColorRampsIds
-      ].map((c) => rgbaStringToComponents(c))
-    )
+  _updateStateWithNewProps = () => {
```

Comment:
I think we can use the [existing update state method](https://deck.gl/docs/api-reference/core/layer#updatestate) to check if `changeFlags` is `propsChanged` so we can set the new state on props change and avoid one re-render